hoodie-hive/pom.xml

```diff
@@ -164,46 +179,90 @@
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-assembly-plugin</artifactId>
-        <version>2.4.1</version>
-        <configuration>
-          <descriptors>
-            <descriptor>src/assembly/src.xml</descriptor>
-          </descriptors>
-          <archive>
-            <manifest>
-              <mainClass>com.uber.hoodie.hive.HiveSyncTool</mainClass>
-            </manifest>
-          </archive>
-
-        </configuration>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>2.5</version>
         <executions>
           <execution>
-            <id>make-assembly</id>
-            <!-- bind to the packaging phase -->
-            <phase>package</phase>
             <goals>
-              <goal>single</goal>
+              <goal>test-jar</goal>
             </goals>
           </execution>
         </executions>
       </plugin>
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-dependency-plugin</artifactId>
+        <artifactId>maven-shade-plugin</artifactId>
         <version>2.4</version>
         <executions>
           <execution>
-            <id>copy-dependencies</id>
             <phase>package</phase>
             <goals>
-              <goal>copy-dependencies</goal>
+              <goal>shade</goal>
             </goals>
             <configuration>
-              <outputDirectory>${project.build.directory}/jars</outputDirectory>
-              <overWriteReleases>false</overWriteReleases>
-              <overWriteSnapshots>false</overWriteSnapshots>
-              <overWriteIfNewer>true</overWriteIfNewer>
+              <relocations>
+                 <relocation>
```

Comment:
mvn shade relies on regex substitution. Using blanket substitutions like "com.", "org." causes unexpected failures at both compile and runtime. Safer way seems to be to shade dependencies in an explicit manner.